apps/admin_app/lib/admin_app_web/controllers/dashboard_controller.ex

```diff
@@ -0,0 +1,96 @@
+defmodule AdminAppWeb.DashboardController do
+  use AdminAppWeb, :controller
+
+  alias Snitch.Data.Model
+
+  def index(conn, params) do
+    {start_date, end_date} = get_date_from_params(params)
+
+    order_state_counts = get_order_state_count(start_date, end_date)
+    product_state_counts = get_product_state_count(start_date, end_date)
+    order_datepoints = get_order_datapoints(start_date, end_date)
+    payment_datapoints = get_payment_datapoints(start_date, end_date)
+
+    render(conn, "index.html",
+      order_state_counts: order_state_counts,
```

Comment:
When the number of assigns is too many it is better to use `put_assign` I think should be used which can be piped one after another.